### PR TITLE
io_lib a new library for SAM/BAM/CRAM

### PR DESCRIPTION
**Pull request by mictadlo**
* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [x] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

**Comment by mictadlo** [comment]
Hi @bioconda/core,
Linux build fine but MacOS has the follwoing problem:
```
https://travis-ci.org/bioconda/bioconda-recipes/jobs/325269580#L768
.deps/libstaden_read_la-cram_io.Tpo -c cram_io.c  -fno-common -DPIC -o .libs/libstaden_read_la-cram_io.o
cram_io.c:66:10: fatal error: 'lzma.h' file not found
#include <lzma.h>
         ^
1 error generated.
make[2]: *** [libstaden_read_la-cram_io.lo] Error 1
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2
BUILD START: io_lib-1.14.9-0
```
Thank you in advance.
Michal

**Comment by mictadlo** [comment]
@bioconda/core  It appears the job did not run on Travis. How to make it run?

**Comment by mictadlo** [comment]
@bioconda/core It appears macos has not started since 16 hours is it normal?

**Comment by druvus** [comment]
Travis had major issues with their osx builds during the last days. It is now fixed but it takes some time to go through the backlogs. Hopefully starts the coming hours.

**Comment by mictadlo** [comment]
@bioconda/core it seems the job on Travis has been canceled. How is it possible to resubmit it again?

**Comment by peterjc** [comment]
A green light from Travis etc back in January.

However, from looking at other pull requests, I think this syntax might not be appropriate anymore:

```
        - zlib {{ CONDA_ZLIB }}*
        - xz {{ CONDA_XZ }}*
```

Is there a GitHub bot to relint / retest a pull request? I can't find it mentioned on https://bioconda.github.io/

**Comment by bgruening** [comment]
We don't have such a fancy system in place currently. Either we close those PRs or ask kindly for a rebase.

**Comment by peterjc** [comment]
Thanks - that's just on conda-forge then (for now). Should those bits now just be:

```
        - zlib
        - xz
```

**Comment by bgruening** [comment]
Yes to both.

File: recipes/io_lib/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+build:
+    number: 0
+    skip: True  # [osx]
+
+requirements:
+    build:
+        - gcc   # [not osx]
+        - zlib {{ CONDA_ZLIB }}*
+        - xz {{ CONDA_XZ }}*
```
Comment (peterjc) [inline recipes/io_lib/meta.yaml]: Please change to just:

```
        - zlib
        - xz
```

File: recipes/io_lib/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+    run:    
+        - libgcc    # [not osx]
+        - zlib {{ CONDA_ZLIB }}*
+        - xz {{ CONDA_XZ }}*
```
Comment (peterjc) [inline recipes/io_lib/meta.yaml]: Again, please change to just:

```
        - zlib
        - xz
```

File: recipes/io_lib/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+test:
+  commands:
+    - echo
+
+about:
+  home: https://sourceforge.net/projects/staden/files/io_lib/ 
+  license: BSD
+  license_file: COPYRIGHT 
+  summary: 'Io_lib is a library of file reading and writing code e.g. for SAM/BAM/CRAM'
```
Comment (peterjc) [inline recipes/io_lib/meta.yaml]: How about `Staden io_lib is a library ...` here?

**Comment by maarten-k** [comment]
another pull request  (#14220)with the same functionality is already pulled.

**Pull request closed without merging**